src/api/libcellml/model.h

```diff
@@ -366,6 +366,89 @@ class LIBCELLML_EXPORT Model: public ComponentEntity
      */
     bool fixVariableInterfaces();
 
+    /**
+     * @brief Add an import source item to this model.
+     *
+     * Add import source by reference to the model. If the import
+     * source item was previously in a different model, it is moved
+     * to this one, and the previous model's import source list is updated.
+     *
+     * The function will return @c false and no action is taken if:
+     *  - The @p imp pointer already exists in this model; or
+     *  - The @p imp is @c nullptr.
+     *
+     * @param imp The import source to add.
+     *
+     * @return @c true if the import source item is added or @c false otherwise.
+     */
+    bool addImportSource(const ImportSourcePtr &imp);
+
+    /**
+     * @brief Get the number of import source items in the model.
```

Comment:
Why "import source items" and not simply "import sources"?

Reply:
Same elsewhere.

Reply:
Because - subtly - they're not the same thing.  We can have one import source, and many import source items.

Reply:
Oops, my bad. I stand corrected!